examples/bayesian_regression.py

```diff
@@ -73,10 +73,12 @@ def model(data):
 
 
 def guide(data):
-    w_loc = torch.randn(1, p, requires_grad=True).type_as(data)
+    w_loc = torch.randn(1, p).type_as(data)
+    w_loc.requires_grad = True
```

Comment:
Looks good, but could you explain the issue so I can avoid it in the future? I thought Adam suggested using the previous form in distributions tests.

Reply:
I just added the comment from @colesbury to the top which best explains this! The issue is that the `.type_as` replaces the earlier `w_mu` which is no longer a leaf node in the computation graph. This behavior has only recently changed, so I'm guessing that Adam's suggestion is from an earlier time.